mixer/adapter/solarwinds/appoptics/batching.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2017 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package appoptics
+
+import (
+	"net/http"
+	"time"
+
+	"bytes"
+
+	"istio.io/istio/mixer/adapter/solarwinds/config"
+	"istio.io/istio/mixer/pkg/adapter"
+)
+
+// BatchMeasurements reads slices of Measurement types off a channel populated by the web handler
+// and packages them into batches conforming to the limitations imposed by the API.
+func BatchMeasurements(loopFactor *bool, prepChan <-chan []*Measurement,
+	pushChan chan<- []*Measurement, stopChan <-chan struct{}, logger adapter.Logger) {
+	var currentBatch []*Measurement
+	dobrk := false
+	ticker := time.NewTicker(time.Millisecond * 500)
+	for *loopFactor {
+		select {
+		case mslice := <-prepChan:
+			if logger.VerbosityLevel(config.DebugLevel) {
+				logger.Infof("AO - batching measurements: %v", mslice)
+			}
+			currentBatch = append(currentBatch, mslice...)
+			if logger.VerbosityLevel(config.DebugLevel) {
```

Comment:
Thoughout this source file, you don't need all these "if" checks around calling the logger.

The only time you need to protect with an if like that, is if the actual call to the logger is expensive, such as if it requires calling fmt.Sprint before invoking the log method.

Reply:
Ok will remove the `if` checks.